src/main/scala/ai/privado/languageEngine/go/passes/config/GoYamlLinkerPass.scala

```diff
@@ -0,0 +1,16 @@
+package ai.privado.languageEngine.go.passes.config
+
+import ai.privado.passes.YamlLinkerPass
+import io.shiftleft.codepropertygraph.generated.Cpg
+import io.shiftleft.codepropertygraph.generated.nodes.Literal
+import io.shiftleft.semanticcpg.language.*
+
+class GoYamlLinkerPass(cpg: Cpg) extends YamlLinkerPass(cpg) {
+  override def matchingLiteralsToPropertyNode(propertyName: String): List[Literal] = {
+    val propertyKey = propertyName.split("\\.").last
+    cpg.literal
+      .codeExact("\"" + propertyKey + "\"")
+      .filter(_.inCall.name("(?i).*getenv").nonEmpty)
+      .toList
```

Comment:
I believe this same kind of thing we are already doing in the Language specific PropertyLinkerPass, didn't understood the usage of this.

Reply:
Actually `javaPropertyLinkerPass` is intended for Java's `.properties` files, which handle a specific use case where we fetch the env value using the `getProperty` call. Yes, we can do this in JavaPropertyLinkerPass also, but handling the YAML related functionality in a separate pass would make it loosely coupled. I had the same discussion with @HiteshMah-Jan regarding this, he also suggested to make separate pass for it.

Reply:
`javaPropertyLinkerPass is intended for Java's .properties files` This statement doesn't hold true, the linking is done irrespective of property being from `.properties` or not

Reply:
I would also recommend seperate pass, but the logic to create a new pass should be on the way the property is linked. Ex - a pass which links `property to the env calls`, instead of a separate linking pass for JSON, YAML

Reply:
> `javaPropertyLinkerPass is intended for Java's .properties files` This statement doesn't hold true, the linking is done irrespective of property being from `.properties` or not

actually its written in the code that `javapropertyLinkerPass` is for `.properties` file here but yes agree what you said, we should have one pass for all similar kind for fetching mechanism irrespective of property file extension. 

I think we also need to do some refactoring, the way current code written is specific to file extension.

Reply:
The comment says so @ankit-privado. The comment is written by David AFAIK, he might not have had the full context of what we were trying to do with the pass. The current pass takes all the properties, and tries to link them. As for the scope of this PR, you can either write your logic inside the JavaPropertyLinker, or do a refactor as @khemrajrathore suggested.

Reply:
yes @karan-batavia, I had a discussion with @khemrajrathore, we will take similar kind of approach to solve this use case in future but as of now we are moving with file extension approach.